htdocs/expensereport/index.php

```diff
@@ -114,32 +112,59 @@
 print '<th colspan="4">'.$langs->trans("Statistics").'</th>';
 print "</tr>\n";
 
-$listoftype=$tripandexpense_static->listOfTypes();
-foreach ($listoftype as $code => $label)
-{
-    $dataseries[]=array($label, (isset($somme[$code])?(int) $somme[$code]:0));
+$listoftype = $tripandexpense_static->listOfTypes();
+$dataseries = array();
+$labels = array();
+foreach ($listoftype as $code => $label) {
+    if (isset($somme[$code])) {
+        $labels[] = html_entity_decode($label);
+        $dataseries[] = (int) $somme[$code];
+    }
 }
 
-if ($conf->use_javascript_ajax)
-{
+if ($conf->use_javascript_ajax) {
     print '<tr><td align="center" colspan="4">';
-
-    include_once DOL_DOCUMENT_ROOT.'/core/class/dolgraph.class.php';
-    $dolgraph = new DolGraph();
-    $dolgraph->SetData($dataseries);
-    $dolgraph->setShowLegend(1);
-    $dolgraph->setShowPercent(1);
-    $dolgraph->SetType(array('pie'));
-    $dolgraph->setWidth('100%');
-    $dolgraph->draw('idgraphstatus');
-    print $dolgraph->show($totalnb?0:1);
+    include_once DOL_DOCUMENT_ROOT.'/core/class/dolchartjs.class.php';
```

Comment:
Why did you change code everywhere we call DolGraph.
The Goal of DolGraph is to be able to change graph library without changing Dolibarr code but just in one place, inside the DolGraph class, so compatibility and/or switching lib can be done easily ?

Reply:
so i think it's on a bad way, i will rework on this later

Reply:
Hope you can find time for this task. 
I spent a lot of time to merge all graphics code into the class DolGraph, so we can now change the library just by changing 1 class (The class DolGraph) , that i hope someone will do it for chartjs (we can even keep use of old lib still possible just by adding a if in this class).